sentry_sdk/api.py

```diff
@@ -156,111 +208,145 @@ def push_scope(  # noqa: F811
     pass
 
 
-@hubmethod
 def push_scope(  # noqa: F811
     callback=None,  # type: Optional[Callable[[Scope], None]]
 ):
     # type: (...) -> Optional[ContextManager[Scope]]
-    return Hub.current.push_scope(callback)
+    """
+    Pushes a new layer on the scope stack.
+
+    :param callback: If provided, this method pushes a scope, calls
+        `callback`, and pops the scope again.
+
+    :returns: If no `callback` is provided, a context manager that should
+        be used to pop the scope again.
+    """
+
+    class _ScopeManager:
+        def __init__(self):
+            # type: () -> None
+            pass
+
+        def __enter__(self):
+            # type: () -> Scope
+            return Scope.get_isolation_scope()
+
+        def __exit__(self, exc_type, exc_value, tb):
+            # type: (Any, Any, Any) -> None
+            pass
+
+    if callback is not None:
+        with push_scope() as scope:
+            callback(scope)
+        return None
+
+    return _ScopeManager()
 
 
 @scopemethod
 def set_tag(key, value):
     # type: (str, Any) -> None
-    return Hub.current.scope.set_tag(key, value)
+    return Scope.get_isolation_scope().set_tag(key, value)
 
 
 @scopemethod
 def set_context(key, value):
     # type: (str, Dict[str, Any]) -> None
-    return Hub.current.scope.set_context(key, value)
+    return Scope.get_isolation_scope().set_context(key, value)
 
 
 @scopemethod
 def set_extra(key, value):
     # type: (str, Any) -> None
-    return Hub.current.scope.set_extra(key, value)
+    return Scope.get_isolation_scope().set_extra(key, value)
 
 
 @scopemethod
 def set_user(value):
     # type: (Optional[Dict[str, Any]]) -> None
-    return Hub.current.scope.set_user(value)
+    return Scope.get_isolation_scope().set_user(value)
 
 
 @scopemethod
 def set_level(value):
     # type: (str) -> None
-    return Hub.current.scope.set_level(value)
+    return Scope.get_isolation_scope().set_level(value)
 
 
-@hubmethod
+@clientmethod
 def flush(
     timeout=None,  # type: Optional[float]
     callback=None,  # type: Optional[Callable[[int, float], None]]
 ):
     # type: (...) -> None
-    return Hub.current.flush(timeout=timeout, callback=callback)
+    return Scope.get_client().flush(timeout=timeout, callback=callback)
 
 
-@hubmethod
+@scopemethod
 def last_event_id():
     # type: () -> Optional[str]
-    return Hub.current.last_event_id()
+    return Scope.get_current_scope().last_event_id()
```

Comment:
Why is this one using the current scope when everything else uses the isolation scope?

Reply:
as the api is removed now, i mark this as resolved.